stdlib/3.4/statistics.pyi

```diff
@@ -0,0 +1,44 @@
+# Stubs for statistics
+# See: http://docs.python.org/3/library/statistics.html
+
+import sys
+from typing import Iterable, Iterator, Sequence, TypeVar, Union, overload
+from decimal import Decimal
+
+# Note: according to the docs, statistics only explicitly supports
+# int, float, Decimal, and Fraction. Other types within the numeric
+# tower are not supported. It also states mixing together different
+# types results in undefined behavior, so the type signatures below
+# deliberately enforce that numeric types may not be mixed.
+#
+# TODO: Once either https://github.com/python/typeshed/pull/545 or
+# https://github.com/python/typeshed/pull/94 is accepted, add support
+# for fractions.
+_TNum = TypeVar('_TNum', int, float, Decimal)
+_T = TypeVar('_T')
+
+def mean(data: Union[Iterator[_TNum], Sequence[_TNum]]) -> _TNum: ...
+if sys.version_info >= (3, 6):
+    def geometric_mean(data: Union[Iterator[_TNum], Sequence[_TNum]]) -> _TNum: ...
+    def harmonic_mean(data: Union[Iterator[_TNum], Sequence[_TNum]]) -> _TNum: ...
+
+# Note: in CPython, the output of median_grouped may sometimes be coerced to
+# float as an implementation detail. In the interests of not breaking code
+# that relies on this implementation detail, the return type is set to the
+# Union of float and the contained numeric type.
+def median(data: Iterable[_TNum]) -> _TNum: ...
```

Comment:
I don't think that the return type of median is right. median() may return the same type as the data _TNum, but if it does an averaging step, it may return a float. For instance:

    >>> statistics.median([2, 4])
    3.0
    >>> statistics.median([2, 4, 6])
    4

I'm not sure how to write that as a type hint. Would that be `Union[_TNum, float]`?

median_low and median_high look correct, as they always return one of the data points and don't interpolate or average between them.